SPDY/NSURLRequest+SPDYURLRequest.m

```diff
@@ -154,4 +188,20 @@ - (void)setSPDYBodyFile:(NSString *)SPDYBodyFile
     }
 }
 
+- (void)setSPDYURLSession:(NSURLSession *)SPDYURLSession
+{
+    [self spdy_setIndirectObject:SPDYURLSession forKey:@"SPDYURLSession"];
+}
+
+- (void)spdy_setIndirectObject:(id)object forKey:(NSString *)key
+{
+    if (object == nil) {
+        [SPDYProtocol removePropertyForKey:key inRequest:self];
```

Comment:
If you store any non-serializable object -- e.g. a reference -- then try to store the NSURLRequest in the NSURLCache, the system spews a runtime warning about "unable to serialize object" (sorry, don't have the exact wording handy). We're trying to avoid that through all these gymnastics. The current way of returning SPDYMetadata was influenced by this same issue. It may be ignorable, but I don't like system warnings.